### ISSUE TITLE
Need help with "Warning: getRouteProps could not find..."

### ISSUE THREAD

**Issue by DevSide** [nozzle/react-static#254]
I've got a problem only on development mode with the client-side navigation. 
I can access `localhost:3000/tom` directly but i can't navigate from `localhost:3000/john` to `localhost:3000/tom` for example.

Any ideas ?

```js
// static.config.js

export default {
  getSiteProps: () => ({}),
  getRoutes: async () => {
    return [
      {
        path: '/john',
        component: 'src/containers/Hi',
        getProps: async () => ({
          name: 'john'
        })
      },
      {
        path: '/tom',
        component: 'src/containers/Hi',
        getProps: async () => ({
          name: 'tom'
        })
      },
    ]
  },
}
```

```js
// src/containers/Hi.js

import React from 'react'
import { getRouteProps } from 'react-static'

export default getRouteProps(({ name }) => (
  <div>
    <h1>Hi {name} !</h1>
  </div>
))
```

**Comment by EmilTholin** [comment]
Hi @DevSide! Could you show us your entry point file ([`src/index.js` in the basic example](https://github.com/nozzle/react-static/blob/master/examples/basic/src/index.js)) and the topmost component file ([`src/App.js` in the the basic example](https://github.com/nozzle/react-static/blob/master/examples/basic/src/App.js)) as well? It's hard to tell what is wrong without seeing those files.

**Comment by DevSide** [comment]
```js
// App.js

import React from 'react'
import { Router, Link } from 'react-static'
//
import Routes from 'react-static-routes'

import './app.css'

export default () => (
  <Router>
    <div>
      <nav>
        <Link key="john" to="/john">john</Link>
        <Link key="tom" to="/tom">tom</Link>
      </nav>
      <div className="content">
        <Routes />
      </div>
    </div>
  </Router>
)
```

```js
// index.js (untouched)

import React from 'react'
import ReactDOM from 'react-dom'
import { AppContainer } from 'react-hot-loader'

// Your top level component
import App from './App'

// Export your top level component as JSX (for static rendering)
export default App

// Render your app
if (typeof document !== 'undefined') {
  const renderMethod = module.hot ? ReactDOM.render : ReactDOM.hydrate
  const render = Comp => {
    renderMethod(
      <AppContainer>
        <Comp />
      </AppContainer>,
      document.getElementById('root'),
    )
  }

  // Render!
  render(App)

  // Hot Module Replacement
  if (module.hot) {
    module.hot.accept('./App', () => {
      render(require('./App').default)
    })
  }
}
```

**Comment by DevSide** [comment]
Looks like because the prefetch is on componentWillMount and not on componentWillReceiveProps. I use the same container ?

**Comment by EmilTholin** [comment]
Very interesting. I tried your example, and there is a bug. I created a identical component `Hi2.js` and used that for one of the routes and it worked. Great find!

**Comment by tannerlinsley** [comment]
Fixed with #255. Release is queued ;)
